pkg/volume/csi/csi_mounter.go

```diff
@@ -314,35 +314,24 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 // 1) if fstype is "", then skip fsgroup (could be indication of non-block filesystem)
 // 2) if fstype is provided and pv.AccessMode == ReadWriteOnly and !c.spec.ReadOnly then apply fsgroup
 func (c *csiMountMgr) applyFSGroup(fsType string, fsGroup *int64) error {
+	klog.Infof("Applying fsGroup %v to %v for volume %v ...", fsGroup, fsType, c.volumeID)
 	if fsGroup != nil {
 		if fsType == "" {
 			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, fsType not provided"))
 			return nil
 		}
-
-		accessModes := c.spec.PersistentVolume.Spec.AccessModes
-		if c.spec.PersistentVolume.Spec.AccessModes == nil {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, access modes not provided"))
-			return nil
-		}
-		if !hasReadWriteOnce(accessModes) {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, only support ReadWriteOnce access mode"))
-			return nil
-		}
-
-		if c.readOnly {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, volume is readOnly"))
-			return nil
-		}
-
-		err := volume.SetVolumeOwnership(c, fsGroup)
-		if err != nil {
-			return err
+		if supportsVolumeOwnership(c.spec) {
+			klog.Infof("Setting ownership for %v (driver = %v)", c.volumeID, c.driverName)
```

Comment:
It's personal choice, but I like to see quotes for long names in inserts. so I'd probably use `%q` for the volumeID.

Reply:
resolving, like you said, personal choice :)